packages/toolpad-app/src/toolpad/AppEditor/PageEditor/MutationEditor.tsx

```diff
@@ -13,79 +13,33 @@ import {
 } from '@mui/material';
 import * as React from 'react';
 import AddIcon from '@mui/icons-material/Add';
-import { BindableAttrEntries, BindableAttrValues, NodeId } from '@mui/toolpad-core';
+import { BindableAttrEntries, NodeId } from '@mui/toolpad-core';
 import invariant from 'invariant';
 import useLatest from '../../../utils/useLatest';
 import { usePageEditorState } from './PageEditorProvider';
 import * as appDom from '../../../appDom';
-import { QueryEditorModel, QueryEditorShellProps } from '../../../types';
+import { QueryEditorModel } from '../../../types';
 import dataSources from '../../../toolpadDataSources/client';
 import NodeNameEditor from '../NodeNameEditor';
 import { update } from '../../../utils/immutability';
 import { useDom, useDomApi } from '../../DomLoader';
 import { ConnectionContextProvider } from '../../../toolpadDataSources/context';
 import ConnectionSelect, { ConnectionOption } from './ConnectionSelect';
-import { createProvidedContext } from '../../../utils/react';
 
 const DATASOURCES_WHITELIST = ['function'];
 
 const EMPTY_OBJECT = {};
 
-interface RenderDialogActions {
-  (params: { isDirty?: boolean; onCommit?: () => void }): React.ReactNode;
-}
-
-interface MutationEditorDialogContext {
-  open: boolean;
-  onClose: () => void;
-  dataSourceId: string | null;
-  renderDialogTitle: () => React.ReactNode;
-  renderDialogActions: RenderDialogActions;
-}
-
-const [useMutationEditorDialogContext, MutationEditorDialogContextProvider] =
-  createProvidedContext<MutationEditorDialogContext>('MutationEditorDialog');
-
-export function MutationEditorShell({ children, isDirty, onCommit }: QueryEditorShellProps) {
-  const { open, onClose, dataSourceId, renderDialogTitle, renderDialogActions } =
-    useMutationEditorDialogContext();
-
-  return (
-    <Dialog fullWidth maxWidth="xl" open={open} onClose={onClose}>
-      {renderDialogTitle()}
-
-      <Divider />
-
-      {dataSourceId ? (
-        <DialogContent
-          sx={{
-            // height will be clipped by max-height
-            height: '100vh',
-            p: 0,
-            display: 'flex',
-            flexDirection: 'column',
-          }}
-        >
-          <Box
-            sx={{
-              flex: 1,
-              minHeight: 0,
-              position: 'relative',
-              display: 'flex',
-            }}
-          >
-            {children}
-          </Box>
-        </DialogContent>
-      ) : (
-        <DialogContent>
-          <Alert severity="error">Datasource &quot;{dataSourceId}&quot; not found</Alert>
-        </DialogContent>
-      )}
+function createQueryModel<Q>(node: appDom.MutationNode<Q>): QueryEditorModel<Q> {
+  const inputParams = node.params || EMPTY_OBJECT;
+  const params =
+    (Object.entries(inputParams).filter(([, value]) => Boolean(value)) as BindableAttrEntries) ||
```

Comment:
NIT: if `key` is not used we could use `Object.values(...)` instead

Reply:
We're operating on entries here, not values, because we want `params` to be a list of entries, not values